File: android/src/main/kotlin/com/ottu/flutter/checkout/CustomerTheme.kt
package com.ottu.flutter.checkout

import kotlinx.serialization.Serializable

@Serializable
data class CustomerTheme(
    val uiMode: String?,
    val mainTitleText: TextStyle?,
    val selectPaymentMethodHeaderText: TextStyle?,
    val titleText: TextStyle?,
    val subtitleText: TextStyle?,
    val feesTitleText: TextStyle?,
    val feesSubtitleText: TextStyle?,
    val dataLabelText: TextStyle?,
    val dataValueText: TextStyle?,
    val errorMessageText: TextStyle?,
    val inputTextField: TextFieldStyle?,
    val sdkBackgroundColor: ColorState?,
    val modalBackgroundColor: ColorState?,
    val selectPaymentMethodHeaderBackgroundColor: ColorState?,
    val paymentItemBackgroundColor: ColorState?,
    val selectorIconColor: ColorState?,
    val savePhoneNumberIconColor: ColorState?,
    val button: ButtonComponent?,
    val backButton: RippleColor?,
    val selectorButton: ButtonComponent?,
    val switchControl: SwitchComponent?,
    val margins: Margins?,
)

@Serializable
data class ColorState(
    val color: String?,
    val colorDisabled: String?,
)

@Serializable
data class RippleColor(
    val color: String?,
    val rippleColor: String?,
    val colorDisabled: String?,
)

@Serializable
data class TextStyle(
    val textColor: ColorState?,
    val fontType: Int?,
)

@Serializable
data class TextFieldStyle(
    val background: ColorState?,
    val primaryColor: ColorState?,
    val focusedColor: ColorState?,
    val text: TextStyle?,
    val error: TextStyle?,
)

@Serializable
data class Margins(
    val left: Int = 0,
    val top: Int = 0,
    val right: Int = 0,
    val bottom: Int = 0,
)

@Serializable
data class ButtonComponent(
    val rippleColor: RippleColor? = null,
    val fontType: Int? = null,
    val textColor: ColorState? = null,
)

@Serializable
data class SwitchComponent(
    val checkedThumbTintColor: String? = null,
    val uncheckedThumbTintColor: String? = null,
    val checkedTrackTintColor: String? = null,
    val uncheckedTrackTintColor: String? = null,
    val checkedTrackDecorationColor: String? = null,
    val uncheckedTrackDecorationColor: String? = null,
)

File: ios/ottu_flutter_checkout/Sources/ottu_flutter_checkout/CustomerTheme.swift
//
//  CustomerTheme.swift
//  ottu_flutter_checkout
//
//  Created by Vi on 29.10.2024.
//

struct CustomerTheme: Decodable {
    let uiMode: String?
    let mainTitle: TextStyle?
    let selectPaymentMethodHeader: TextStyle?
    let title: TextStyle?
    let subtitle: TextStyle?
    let feesTitle: TextStyle?
    let feesSubtitle: TextStyle?
    let dataLabel: TextStyle?
    let dataValue: TextStyle?
    let errorMessage: TextStyle?
    let inputTextField: TextFieldStyle?
    let backgroundColor: ColorState?
    let modalBackgroundColor: ColorState?
    let selectPaymentItemTitleBackgroundColor: ColorState?
    let paymentItemBackgroundColor: ColorState?
    let paymentItemDescriptionTextColor: ColorState?
    let paymentItemBorderColor: ColorState?
    let paymentItemBorderWidth: Float?
    let paymentItemCornerRadius: Float?
    let selectorIconColor: ColorState?
    let savePhoneNumberIconColor: ColorState?
    let button: ButtonCmt?
    let backButton: RippleColor?
    let selectorButton: ButtonCmt?
    let switchControl: SwitchComponent?
    let margins: Margins?
}

struct ColorState: Decodable {
    let color: String?
    let colorDisabled: String?
}

struct RippleColor: Decodable {
    let color: String?
    let rippleColor: String?
    let colorDisabled: String?
}

struct TextStyle: Decodable {
    let textColor: ColorState?
    let fontFamily: String?
    let fontSize: Int?
}

struct TextFieldStyle: Decodable {
    let background: ColorState?
    let primaryColor: ColorState?
    let focusedColor: ColorState?
    let text: TextStyle?
    let error: TextStyle?
}

struct Margins: Decodable {
    var left: Int = 0
    var top: Int = 0
    var right: Int = 0
    var bottom: Int = 0
}

struct ButtonCmt: Decodable {
    let rippleColor: RippleColor?
    let fontFamily: String?
    let textColor: ColorState?
    let textDisabledColor: ColorState?
    let borderColor: ColorState?
    let borderWidth: Float?
    let cornerRadius: Float?
}

struct SwitchComponent: Decodable {
    let checkedThumbTintColor: String?
    let uncheckedThumbTintColor: String?
    let checkedTrackTintColor: String?
    let uncheckedTrackTintColor: String?
    let checkedTrackDecorationColor: String?
    let uncheckedTrackDecorationColor: String?
}
